src/decision/location_selection.jl

```diff
@@ -0,0 +1,356 @@
+
+"""
+    _site_selection(domain::Domain, mcda_vars::DMCDA_vars, guided::Int64)
+
+Perform site selection using a chosen aggregation method, domain, initial cover, criteria weightings and thresholds.
+
+# Arguments
+- `domain` : ADRIA Domain type, indicating geographical domain to perform site selection over.
+- `mcda_vars` : Contains relevant parameters for performing guided location selection.
+- `guided` : Integer indicating aggegation algorithm to use for guided location selection.
+
+# Returns
+- `ranks` : n_reps * sites * 3 (last dimension indicates: site_id, seeding rank, shading rank)
+    containing ranks for single scenario.
+"""
+function _site_selection(domain::Domain, 
+    mcda_vars::DMCDA_vars, 
+    guided::Int64)
+    
+    site_ids = mcda_vars.site_ids
+    n_sites = length(site_ids)
+
+    # site_id, seeding rank, shading rank
+    rankingsin = [mcda_vars.site_ids zeros(Int64, n_sites) zeros(Int64, n_sites)]
+
+    prefseedsites::Vector{Int64} = zeros(Int64, mcda_vars.n_site_int)
+    prefshadesites::Vector{Int64} = zeros(Int64, mcda_vars.n_site_int)
+
+    # Determine connectivity strength
+    # Account for cases where no coral cover
+    in_conn, out_conn, strong_pred = connectivity_strength(domain.TP_data .* site_k_area(domain), Array(mcda_vars.sum_cover))
```

Comment:
I see you changed the expected type in `connectivity_strength()` for this.

I'm just wondering why an array is expected here. Isn't vector more appropriate?

Reply:
I can't remember exactly why I did this but I believe it was necessary to fit both the ecological model and standalone site selection cases... but I'll get back to you on this

Reply:
I've changed the types tp be more specific now and work for both the ecological model and standalone case

Reply:
> I've changed the types tp be more specific now and work for both the ecological model and standalone case

I'd then have to ask why they have to use different types?

Would this be resolved simply by passing in a `vec` instead?

Reply:
For the standalone site selection the second argument is Vector{Float32} whereas for the ecological model it is Vector{Float64}

Reply:
`collect()` is what you want.